include/seqan3/core/algorithm/configuration_element_parallel_mode.hpp

```diff
@@ -25,9 +26,32 @@ namespace seqan3::detail
  * This type is used to enable the parallel mode of the algorithms.
  */
 template <typename wrapped_config_id_t>
-struct parallel_mode : public pipeable_config_element<parallel_mode<wrapped_config_id_t>, uint32_t>
+class parallel_mode : public pipeable_config_element<parallel_mode<wrapped_config_id_t>>
 {
-    //!\brief Internal id to check for consistent configuration settings.
+public:
+    /*!\name Constructors, assignment and destructor
+     * \{
+     */
+    parallel_mode() = default; //!< Defaulted.
+    parallel_mode(parallel_mode const &) = default; //!< Defaulted.
+    parallel_mode(parallel_mode &&) noexcept = default; //!< Defaulted.
+    parallel_mode & operator=(parallel_mode const &) = default; //!< Defaulted.
+    parallel_mode & operator=(parallel_mode &&) noexcept = default; //!< Defaulted.
```

Comment:
```suggestion
    parallel_mode(parallel_mode &&) = default; //!< Defaulted.
    parallel_mode & operator=(parallel_mode const &) = default; //!< Defaulted.
    parallel_mode & operator=(parallel_mode &&) = default; //!< Defaulted.
```
or did you define this on purpose? because usually for `= default` constructors this is added automatically.

Reply:
I added `noexcept` on purpose, because when I left it away, CLion told me:
```
Clang-Tidy: Move constructors should be marked noexcept.
Clang-Tidy: Move assignment operators should be marked noexcept.
```

If it's our policy I can remove them now.

Reply:
Yes :)
See here: https://en.cppreference.com/w/cpp/language/noexcept_spec that they are noexcept if they can.